maintenance/wikia/TemplateClassificationRecognizedMetric.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+require_once __DIR__ . '/../Maintenance.php';
+
+/**
+ * Maintenance script counts current number of recognized templates in content namespaces
+ *
+ * @author Kamil Koterba <kamil@wikia-inc.com>
+ * @ingroup Maintenance
+ */
+class TemplateClassificationRecognizedMetricMaintenance extends Maintenance {
+
+	/**
+	 * Set script options
+	 */
+	public function __construct() {
+		parent::__construct();
+
+		$this->mDescription = 'Maintenance script counts current number of recognized templates in content namespaces';
+	}
+
+	/**
+	 * Script entry point
+	 */
+	public function execute() {
+		global $wgContentNamespaces, $wgDWStatsDB;
+
+		$statsDBRead = wfGetDB( DB_SLAVE, [], $wgDWStatsDB );
+		$wamTop500 = $this->getWamTop500( $statsDBRead, ( new \WikiaSQL ) );
+
+		$top500CNRecognizedCount = 0;
+		$top500CNAllCount = 0;
+
+		foreach ( $wamTop500 as $wikia ) {
+			$wikiaDBName = WikiFactory::IDtoDB( $wikia->wiki_id );
+			$wikiaDBRead = wfGetDB( DB_SLAVE, [], $wikiaDBName );
+
+			$cnTemplatesOnWikia = $this->getContentNamespacesTemplates( $wikiaDBRead, $wgContentNamespaces );
+			$recognizedTemplates  = $this->getRecognizedTemplatesOnWikia(
+				( new TemplateClassificationService() ),
+				$wikia->wiki_id
+			);
+
+			$cnRecognizedTemplates = $this->intersectSets( $cnTemplatesOnWikia, $recognizedTemplates );
+
+			$top500CNRecognizedCount += count( $cnRecognizedTemplates );
+			$top500CNAllCount += count( $cnTemplatesOnWikia );
+		}
+
+		/*
+		 * Store value in DB
+		 * TODO uncomment and test when storage is ready
+		 */
+//		$statsDBWrite = wfGetDB( DB_MASTER, [], $wgStatsDB ); // Change DB
+//		$timestamp = date( 'Y-m-d' );
+//		$this->setStats( $top500CNRecognizedCount, $top500CNAllCount, $timestamp, $statsDBWrite );
+
+		$this->pushGecko( $top500CNRecognizedCount, $top500CNAllCount );
+	}
+
+	private function getRecognizedTemplatesOnWikia( TemplateClassificationService $tcService, $wikiaId ) {
+		$templates = $tcService->getTemplatesOnWiki( $wikiaId );
+		foreach ( $templates as $pageId => $type ) {
+			if ( !$this->isRecognized( $type ) ) {
+				unset( $templates[$pageId] );
+			}
+		}
+		return $templates;
+	}
+
+	private function isRecognized( $type ) {
+		return $type !== TemplateClassificationService::TEMPLATE_UNKNOWN
```

Comment:
Frankly, this would probably read easier given the logic inside if it were 
privat function isUnrecognized( $type ) {
return in_array($type,[TemplateClassificationService::TEMPLATE_UNKNOWN,TemplateClassificationService::TEMPLATE_UNCLASSIFIED, TemplateClassificationService::TEMPLATE_OTHER]);
}